relay-general/src/store/normalize/contexts.rs

```diff
@@ -199,12 +204,44 @@ fn normalize_response(response: &mut ResponseContext) {
     }
 }
 
-pub fn normalize_context(context: &mut Context) {
+static ANDROID_MAP: OnceCell<Result<HashMap<String, String>, std::io::Error>> = OnceCell::new();
+
+fn get_android_product_name(model: &str) -> Option<String> {
+    let mymap: &Result<HashMap<String, String>, std::io::Error> = ANDROID_MAP.get_or_init(|| {
+        let path = PathBuf::from("../android_models.csv");
+        let file = std::fs::File::open(&path)?;
+        let reader = std::io::BufReader::new(file);
+        let mut android_map: HashMap<String, String> = HashMap::new();
+
+        for line in reader.lines() {
+            let line = line?;
+            let columns: Vec<&str> = line.split(',').collect();
+
+            if columns.len() >= 4 {
+                let key = columns[3].trim().to_string();
+                let value = columns[1].trim().to_string();
+                android_map.insert(key, value);
+            }
+        }
+        Ok(android_map)
+    });
+
+    mymap.as_ref().ok().and_then(|map| map.get(model).cloned())
+}
+
+pub fn normalize_context(context: &mut Context, config: &StoreConfig) {
     match context {
         Context::Runtime(runtime) => normalize_runtime_context(runtime),
         Context::Os(os) => normalize_os_context(os),
         Context::Response(response) => normalize_response(response),
-        _ => (),
+        Context::Device(device) => {
+            if let Some(model) = device.as_ref().model.value() {
+                if let Some(product_name) = get_android_product_name(model) {
+                    device.model.set_value(Some(product_name));
+                }
```

Comment:
nit: if possible, I'd use [`and_then`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then) here, I think it's more readable. Something like:
```suggestion
                get_android_product_name(model).and_then(|product_name| device.model.set_value(Some(product_name)));
                }
```